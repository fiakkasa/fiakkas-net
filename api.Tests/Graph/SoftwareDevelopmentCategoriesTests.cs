using HotChocolate.Execution;

namespace api.Tests.Graph;

public partial class SoftwareDevelopmentCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task SoftwareDevelopmentCategories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              softwareDevelopmentCategories {
                totalCount
                nodes {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                  portfolioCategories {
                    totalCount
                  }
                  portfolioCustomers {
                    totalCount
                  }
                  portfolioItems {
                    totalCount
                  }
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task SoftwareDevelopmentCategoryById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              softwareDevelopmentCategoryById(id: "U29mdHdhcmVEZXZlbG9wbWVudENhdGVnb3J5CmdjYTgzMmJmOWI3Y2I0YzMxYmY4ZDAwZjg3YTI3NmZlMw==") {
                createdAt
                href
                id
                internalId
                title
                updatedAt
                version
              }
            }
            """);

        var fn = result.ExpectOperationResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
