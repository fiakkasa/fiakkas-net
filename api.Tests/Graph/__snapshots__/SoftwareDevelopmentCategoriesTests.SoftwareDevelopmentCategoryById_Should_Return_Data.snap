{
  "data": {
    "softwareDevelopmentCategoryById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "href": "/test",
      "id": "U29mdHdhcmVEZXZlbG9wbWVudENhdGVnb3J5Ovkrg8rLtzFMv40A+Honb+M=",
      "internalId": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
